CondFormats/HcalObjects/src/HcalPedestalsGPU.cc

```diff
@@ -0,0 +1,83 @@
+#include "CondFormats/HcalObjects/interface/HcalPedestalsGPU.h"
+
+#include "CondFormats/HcalObjects/interface/HcalPedestals.h"
+
+#include "FWCore/Utilities/interface/typelookup.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+
+// FIXME: add proper getters to conditions
+HcalPedestalsGPU::HcalPedestalsGPU(HcalPedestals const& pedestals)
+    : unitIsADC_{pedestals.isADC()},
+      totalChannels_{pedestals.getAllContainers()[0].second.size() + pedestals.getAllContainers()[1].second.size()},
```

Comment:
Let me make a note on a semi-random place. The `HcalPedestals::getAllContainers()` seems to have lot's of overhead (constructing `vector<pair<string, vector<HcalPedestal>>>` on each call). I wonder if this could be (eventually) improved, event if the price is paid only at IOV changes.